src/commands/registerCommands.ts

```diff
@@ -181,22 +182,83 @@ async function grammarAssociationCommand(documentURI: Uri, languageClient: Langu
     // The XSD, DTD has been selected, get the proper syntax for binding this grammar file in the XML document.
     const identifier = TextDocumentIdentifier.create(documentURI.toString());
     const grammarURI = fileUri[0];
+    const grammarFilename = path.basename(grammarURI.toString())
+    const currentFilename = path.basename(identifier.uri);
     try {
-      const result = await commands.executeCommand(ServerCommandConstants.ASSOCIATE_GRAMMAR_INSERT, identifier, grammarURI.toString(), bindingType);
-      // Insert the proper syntax for binding
-      const lspTextDocumentEdit = <TextDocumentEdit>result;
-      const workEdits = new WorkspaceEdit();
-      for (const edit of lspTextDocumentEdit.edits) {
-        workEdits.replace(documentURI, languageClient.protocol2CodeConverter.asRange(edit.range), edit.newText);
+      if (bindingType == `fileAssociation`) {
+        // Bind grammar using file association
+        await bindWithFileAssociation(grammarFilename, currentFilename);
+      } else {
+        // Bind grammar using standard binding
+        await bindWithStandard(identifier, grammarURI, bindingType, documentURI, languageClient);
```

Comment:
Please move identifier declaration inside bindWithStandard

Reply:
I believe the `currentFileName` requires the identifier. Should I also move it into `bindWithFileAssociation`, or would it be better if I replaced the currentFileName param with the identifier?

Reply:
identifier is created from documentURI that bindWithStandard have as parameter. S

replace with 

`await bindWithStandard( grammarURI, bindingType, documentURI, languageClient);`

and inside  bindWithStandard:

`const identifier = TextDocumentIdentifier.create(documentURI.toString());`